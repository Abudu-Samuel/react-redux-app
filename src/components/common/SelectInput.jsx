/* eslint-disable jsx-a11y/label-has-for */
import React from 'react';
import PropTypes from 'prop-types';

const SelectInput = ({
  name, label, onChange, defaultOption, defaultValue, error, options
}) => (
  <div className="form-group">
    <label htmlFor={name}>{label}</label>
    <div className="field">
      <select
          name={name}
          className="form-control"
          onChange={onChange}
          defaultValue={defaultValue}
        >
        <option value="">{defaultOption}</option>
        {options.map(option => <option key={option.value} value={option.value}>{option.text}</option>)}
      </select>
      {error && <div className="alert alert-danger">{error}</div>}
    </div>
  </div>
);


SelectInput.propTypes = {
  name: PropTypes.string.isRequired,
  label: PropTypes.string.isRequired,
  onChange: PropTypes.func.isRequired,
  defaultOption: PropTypes.string.isRequired,
  defaultValue: PropTypes.string,
  error: PropTypes.string,
  options: PropTypes.arrayOf(PropTypes.object).isRequired
};

export default SelectInput;