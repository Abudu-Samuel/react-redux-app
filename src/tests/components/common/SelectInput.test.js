/* eslint-disable */
import SelectInput from '../../../components/common/SelectInput';

const config = (error) => {
  const props = {
    name: '',
    label: '',
    onChange: jest.fn(),
    defaultOption: '',
    defaultValue: '',
    error,
    options: [{ value: '', text: '' }]
  };
  return mount(<SelectInput {...props} />);
};

let mountWrap;

describe('<SelectInput />', () => {
  it('should render SelectInput component', () => {
    mountWrap = config('');
    expect(toJson(mountWrap)).toMatchSnapshot();
  });

  it('should render error tag when an error occurs', () => {
    mountWrap = config('select an author');
    expect(mountWrap.find('.alert-danger').length).toEqual(1);
    expect(mountWrap.find('.alert-danger').get(0).props.children).toEqual('select an author');
  });
});