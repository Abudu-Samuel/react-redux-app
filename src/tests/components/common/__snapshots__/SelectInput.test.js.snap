// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SelectInput /> should render SelectInput component 1`] = `
<SelectInput
  defaultOption=""
  defaultValue=""
  error=""
  label=""
  name=""
  onChange={[MockFunction]}
  options={
    Array [
      Object {
        "text": "",
        "value": "",
      },
    ]
  }
>
  <div
    className="form-group"
  >
    <label
      htmlFor=""
    />
    <div
      className="field"
    >
      <select
        className="form-control"
        defaultValue=""
        name=""
        onChange={[MockFunction]}
      >
        <option
          value=""
        />
        <option
          key=""
          value=""
        />
      </select>
    </div>
  </div>
</SelectInput>
`;
